### DISCUSSION TITLE
Event handler for after_contribution_accept

### DISCUSSION THREAD

**Discussion by jingyunc** [NVIDIA/NVFlare#2076] [Q&A] [answered]
Hello, I'm looking for the code piece where the event AFTER_CONTRIBUTION_ACCEPT is handled.

This event was fired in the Swarm Learning controller example (swarm_client_ctl.py). But I didn't find the code piece where the event was handled. Maybe the handle_event() was left undefined for user to customize?

Thanks in advance.

**Comment by SYangster** [comment, marked as answer]
Right, this specific AFTER_CONTRIBUTION_ACCEPT event has been predefined, but is not currently handled. While many common events are handled throughout code logic, we provide all of these event definitions in the case that they could be useful. 

Feel free to write your own handle_event() for this event. (also see the [docs](https://nvflare.readthedocs.io/en/main/programming_guide/event_system.html) for more detail)
